CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVReader.java

```diff
@@ -0,0 +1,114 @@
+package io.vantiq.extsrc.CSVSource;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+
+import io.vantiq.extjsdk.ExtensionWebSocketClient;
+
+public class CSVReader {
+    private static final String MAX_LINES_IN_EVENT = "maxLinesInEvent";
+
+
+    static void sendNotification(String filename, int numPacket , ArrayList<Map<String,String>> file ,   ExtensionWebSocketClient oClient )
+    {
+        Map<String,Object> m = new LinkedHashMap<>();
+        m.put("file", filename);
+        m.put("segment", numPacket);
+        m.put("lines", file.toArray());
+        
+        
+        oClient.sendNotification(m);
+    }
+    static String setFieldName(int i , Map<String,String>schema)
+    {
+        String field = String.format("field%d",i); 
+        if (schema != null)
+        {
+            if (schema.get(field)!= null)
+            {
+                field = schema.get(field) ; 
+            }
+        }
+
+        return field ; 
+    }
+    @SuppressWarnings("unchecked")
+    static public ArrayList<Map<String,String>> execute(String csvFile,  Map<String, Object>     config,   ExtensionWebSocketClient oClient  ) {
+
+        String line = "";
+        int numOfRecords ; 
+        int numOfProcessedLinesInSegment = 0 ; 
+        Map<String,String> schema = null;
+
+        if (config.get("schema") != null)
+        {
+            schema = (Map<String,String>) config.get("schema");
+        }
+
+        String delimiter = ";";
```

Comment:
I still think that if we're calling this the Comma Separated Values extension (which is what CSV stands for), the default value for the delimiter should be ",".

Also, should have a test for it being set to something else, for the default, and for it's being set to a multi-character RegEx.  _E.g._ ... "delimiter" : "[0-9]*" should turn "fred12345namir33marty" into [fred, namir, marty]. Since we advertise this (presumably), we should ensure that it works.

Reply:
using [0-9] gives much more meaningful output , i add support for that , but i currently don't see it in practice.